### DISCUSSION TITLE
Cycle through clients

### DISCUSSION THREAD

**Discussion by aba-wtag** [nikitabobko/AeroSpace#716] [questions-and-answers]
I am new to `AeroSpace` previously I have used i3WM, DWM, HyperLand and a big fan of DWM. 

In DWM there is a way of cycle through the opened clients by pressing only `J` or `K` key. I am wondering is there something similar here also or how can I achieve this functionality?

**Comment by alextremblay** [comment]
have experience with i3 but not DWM. Not sure what is a client in this context?

**Reply by aba-wtag** [reply]
Client means open application instances. For example in the below picture three instances (i.e. clients) of `WezTerm` are open.
![Screenshot 2024-11-21 at 10 25 40 AM](https://github.com/user-attachments/assets/52a8ba3f-5725-46f6-b658-9546b473ad0e)

**Comment by alextremblay** [comment]
Ahhhhh, ok. In Aerospace (and MacOS in general) we call these windows

I generally only use accordion layout so i can't say for sure, but mapping a key to `focus right --boundaries-action wrap-around-the-workspace` might do what you want.

If not, you could write a script that:
 1. counts the number of windows in the currect workspace (something like `COUNT=$(aerospace list-windows --workspace focused | wc -l)`
 2. cycles through a number from 0 to `COUNT`
 3. executes `aerospace focus --dfs-index <that number>`

Not sure how you would manage state (current index number to increment per script invocation), but i suppose that must be left as an exercise for the reader :)


Or, it's entirely possible that there's another simpler solution... not sure

**Comment by gldtn** [comment]
If you mean focus and copied the default config: https://nikitabobko.github.io/AeroSpace/guide#default-config

```toml
# See: https://nikitabobko.github.io/AeroSpace/commands#focus
alt-h = 'focus left'
alt-j = 'focus down'
alt-k = 'focus up'
alt-l = 'focus right'
```

**Comment by alextremblay** [comment]
I think @aba-wtag is looking for the ability to cycle through all available windows in a workspace, not the ability to focus the next window in a given direction.

Something more like #248

**Reply by gldtn** [reply]
Ahhh I see. Yeah, yabai(yabai -m space --rotate|--mirror) has something similar! Pretty neat feature to have.
